programs/magicblock/src/utils/account_actions.rs

```diff
@@ -14,8 +14,10 @@ pub(crate) fn set_account_owner(
     acc.borrow_mut().set_owner(pubkey);
 }
 
+/// Sets proper values on account during undelegation
 pub(crate) fn set_account_owner_to_delegation_program(
     acc: &RefCell<AccountSharedData>,
 ) {
+    acc.borrow_mut().set_delegated(false);
```

Comment:
We should not do that here IMO.
Instead once the acocunt updates on chain and is cloned again we should change that flag since then the delegation record would no longer be there.
Is there a reason for this change? The whole cloning pipeline will be replaced soon, so ideally we'd leave the obsolete one as is to not run into more issues until then.

At any rate we'd have to rename that function if we do change the behavior.

Reply:
Resolving as this change will be in separate pr